### PR TITLE
bug: prevent violent error when Experiment metadata is missing

### PR DESCRIPTION
**Pull request by stephanwlee**
ui_selector which provides a convenience selector for getting the run
selection was throwing a JavaScript exception violently when experiment
metadata did not load but runs list was loaded.

Above case can happen when the backend implements /data/experiments
incorrectly and inadvertently apply `limit` on the list of experiments
passed. As such, TensorBoard may not fetch an experiment metadata ever
when list of experiment ids go above `N`.

In theory, there should be a fix to the backend, but we also do not want
the web code to violently throw and henceforth break the app completely.
Especially since the web code never makes use of the experiment name
anymore when filtering, we can easily guard against the case, hence the
change.

### CODE REVIEW

File: tensorboard/webapp/util/ui_selectors_test.ts [hunk 1]
```diff
@@ -265,6 +265,63 @@ describe('ui_selectors test', () => {
           ])
         );
       });
+
+      it('does not violently throw when an experiment metadata DNE', () => {
```
Comment (bmd3k) [inline tensorboard/webapp/util/ui_selectors_test.ts:269]: What is DNE?

**Review by bmd3k — APPROVED** [review]

File: tensorboard/webapp/util/ui_selectors_test.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/util/ui_selectors_test.ts:269]: Does not exist. It was terminology I used when doing math. If it is too obscure, I will change it. Let me know what you prefer.

Comment (bmd3k) [reply tensorboard/webapp/util/ui_selectors_test.ts:269]: Yes it seems a little obscure.

Comment (stephanwlee) [reply tensorboard/webapp/util/ui_selectors_test.ts]: Done.

**Pull request merged**